### PR TITLE
(Feature) Add validation rules for CRDs

### PR DESCRIPTION
**Pull request by nikita-vanyasin**

### CODE REVIEW

File: internal/cr_validation_test.go [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+
+	if len(docName) > 0 {
+		s.ExternalDocs = &apiextensions.ExternalDocumentation{
+			URL: getDocsLinkForField(docName, def),
+		}
+	}
+	s.Description = strings.Join(def.Docs, "\n")
+}
+
+func Test_GenerateCRValidationSchemas(t *testing.T) {
```
Comment (jwierzbo) [inline internal/cr_validation_test.go:70]: We should add a comment here that we used that test to generate schema files

Comment (nikita-vanyasin) [reply internal/cr_validation_test.go:71]: done

**Review by jwierzbo — DISMISSED** [review]

**Review by jwierzbo — CHANGES_REQUESTED** [review]
Please add doc for that feature under doc/features

**Comment by nikita-vanyasin** [comment]
> Please add doc for that feature under doc/features

Done. See `crds.md`
I've added it to docs root because `feature` contains only features generated from features.yaml - which is not the case for CRD validation.

**Review by jwierzbo — APPROVED** [review]

File: pkg/crd/crds/backups-backup.schema.generated.json [hunk 2]
```diff
@@ -0,0 +1,208 @@
+{
```
Comment (ajanikow) [inline pkg/crd/crds/backups-backup.schema.generated.json:1]: Try to parse it to yaml - use k8s.io/apimachinery/pkg/util/yaml

File: pkg/crd/crds/backups-backuppolicy.go [hunk 3]
```diff
@@ -34,23 +33,26 @@ const (
...
-	if err := yaml.Unmarshal(backupsBackupPolicy, &backupsBackupPolicyCRD); err != nil {
-		panic(err)
-	}
+	mustLoadCRD(backupsBackupPolicy, backupsBackupPolicySchemaRaw, &backupsBackupPolicyCRD, &backupsBackupPolicyCRDWithSchema)
 }
 
-func BackupsBackupPolicyPolicy() *apiextensions.CustomResourceDefinition {
-	return backupsBackupPolicyCRD.DeepCopy()
+func BackupsBackupPolicyPolicy(opts ...GetCRDOptions) *apiextensions.CustomResourceDefinition {
+	return getCRD(backupsBackupPolicyCRD, backupsBackupPolicyCRDWithSchema, opts...)
```
Comment (ajanikow) [inline pkg/crd/crds/backups-backuppolicy.go:40]: Just unmarshal it twice instead of creating new func

Comment (ajanikow) [reply pkg/crd/crds/backups-backuppolicy.go:40]: BackupsBackupPolicyPolicy -> this should have same working function, add BackupsBackupPolicyPolicyWithOptions

Comment (ajanikow) [reply pkg/crd/crds/backups-backuppolicy.go:40]: BackupsBackupPolicyPolicyWithOptions should default options to the one which is used by CRD -> for ML True, others False (if not set at all)

**Review by ajanikow — CHANGES_REQUESTED** [review]

File: pkg/crd/crds/backups-backup.schema.generated.json (hunk 2 above)
Comment (nikita-vanyasin) [reply pkg/crd/crds/backups-backup.schema.generated.json]: Done

**Comment by nikita-vanyasin** [comment]
Will split PR into two smaller

**Pull request closed without merging**